Identity ga doc updates (#36304)
Azure/azure-sdk-for-java: sdk/identity/azure-identity/TOKEN_CACHING.md

**Note:** When Azure Identity library credentials are used with Azure service libraries (for example, Azure Blob Storage), the in-memory token caching is active in the `HttpPipeline` layer as well. All `TokenCredential` implementations are supported there, including custom implementations external to the Azure Identity library.

#### Caching cannot be disabled

> As there are many levels of cache it is not possible disable in-memory caching. The in-memory cache may be cleared by creating a new credential instance.

Reviewer: ```suggestion

As there are many levels of cache, it's not possible disable in-memory caching. However, the in-memory cache may be cleared by creating a new credential instance.
```

- Ensures the app can continue to function during an Azure AD outage or disruption.
- Avoids having to prompt users to authenticate each time the process is restarted.

> 

Reviewer 2: Consider: add the missing section on clearing the token cache

Author: added a line. Not sure if it needs to be more than that.

#### Implementation Note

> The cache is broken into two files to allow for separate caching of Continuous Access Evaluation (CAE) vs non-CAE credentials. The default filename is `msal` but can be specified through `TokenCachePersistenceOptions.setName()`. The filenames end in `.cae` or `.nocae` respectively.

Reviewer 2: nit: I’d use phrasing other than broken - like split

- Ensures the app can continue to function during an Azure AD outage or disruption.
- Avoids having to prompt users to authenticate each time the process is restarted.

#### Implementation Note

> The cache is stored in two files to allow for separate caching of Continuous Access Evaluation (CAE) vs non-CAE credentials. The default filename is `msal` but can be specified through `TokenCachePersistenceOptions.setName()`. The filenames end in `.cae` or `.nocae` respectively.

Reviewer 3: And this isn't the whole story because we cache in files only on Windows and when falling back to plaintext on other platforms. I think we should just leave all this as undocumented implementation details i.e., delete this section. What should customers do with this information whose continued correctness we won't guarantee?

> The cache is stored in two files to allow for separate caching of Continuous Access Evaluation (CAE) vs non-CAE credentials. The default filename is `msal` but can be specified through `TokenCachePersistenceOptions.setName()`. The filenames end in `.cae` or `.nocae` respectively.

Author: Yeah, I'm torn on this. There was some discussion in one of our meetings about this around customers who might have somehow taken a dependency on the existing file. I am fine to remove it from here. @joshfree, I think this was one you had opinions about. (At least you were the last one to say something before I wrote it down.)

> The cache is stored in two files to allow for separate caching of Continuous Access Evaluation (CAE) vs non-CAE credentials. The default filename is `msal` but can be specified through `TokenCachePersistenceOptions.setName()`. The filenames end in `.cae` or `.nocae` respectively.

Author: Removed the section after discussion.
